src/libraries/Microsoft.Extensions.Caching.Memory/src/CacheEntry.cs

```diff
@@ -218,28 +276,25 @@ private static void ExpirationTokensExpired(object obj)
 
         internal void InvokeEvictionCallbacks() => _tokens?.InvokeEvictionCallbacks(this);
 
-        // this simple check very often allows us to avoid expensive call to PropagateOptions(CacheEntryHelper.Current)
-        [MethodImpl(MethodImplOptions.AggressiveInlining)] // added based on profiling
-        internal bool CanPropagateOptions() => (_tokens != null && _tokens.CanPropagateTokens()) || AbsoluteExpiration.HasValue;
+        internal bool CanPropagateTokens() => _tokens != null && _tokens.CanPropagateTokens();
```

Comment:
Why did you move the `AbsoluteExpiration` check out of this method?

Reply:
If I remember correctly, then this is for better codegen (the OR condition together with a not-zero check ended up with quite inefficient codegen).